src/Vision/grippyCode.py

```diff
@@ -0,0 +1,168 @@
+import gripV2
+import cv2
+import numpy as np
+import threading
+import math
+from networktables import NetworkTables
+
+
+cond = threading.Condition()
+notified = [False]
+
+def connectionListener(connected, info):
+    print(info, '; Connected=%s' % connected)
+    with cond:
+        notified[0] = True
+        cond.notify()
+
+NetworkTables.initialize(server='10.1.66.2')
+NetworkTables.addConnectionListener(connectionListener, immediateNotify=True)
+table = NetworkTables.getTable('Vision Correction Table')
+with cond:
+    if not notified[0]:
+        cond.wait()
+
+cap = cv2.VideoCapture(1)
+cap.set(10,30)
+
+goalFinder = gripV2.GripiPipeline()
+
+rightAngle = -15
+leftAngle = -75
+deadzone = 15
+
+
+def findPairOffset(pair):
+    leftcX = pair[0][0][0]
+    rightcX = pair[1][0][0]
+    pairMidpoint = (leftcX + rightcX) / 2
+    pairOffset = abs(imageMidpoint - pairMidpoint)
+    return pairOffset
+
+#First Search for the -15 degree contour and add to first part of pair,
+#then search for -75 degree contour, put in pair, assure the -75 contour is to the right of the -15 degree contour,
+#then assure the -75 contour is the closest contour to the -15,
+#by searching for other -75 contours that are either closer or farther
+#DONEEEEEEEE!!!!!!!!
+#   (find center of each pair then identify which pair is closest to image center)
+def findPairs(contourList):
+    rightRectList = []
+    leftRectList = []
+    rectPairList = []
+
+    #print ("Number of Rectangles: {}".format(len(contourList)))
+
+    for contour in contourList:
+        rectangleBox = cv2.minAreaRect(contour)
+ 
+        if rectangleBox[2] > rightAngle - deadzone and rectangleBox[2] < rightAngle + deadzone:
+            rightRectList.append(rectangleBox)
+       
+        if rectangleBox[2] > leftAngle - deadzone and rectangleBox[2] < leftAngle + deadzone:
+            leftRectList.append(rectangleBox) 
+
+    for leftRect in leftRectList:
+        pair = (leftRect,None)
+        for rightRect in rightRectList:
+            #0 is center and 0 is x coordinate
+            if rightRect[0][0] > leftRect[0][0]:
+                if pair[1] is None or rightRect[0][0] < pair[1][0][0]:
+                    pair = (leftRect,rightRect)
+        if pair[1] != None:
+            rectPairList.append(pair)
+
+    closestCenterPair = None
+    for pair in rectPairList:
+        print ("rectangle pair coordinates: {}, {}".format(pair[0][0][0],pair[1][0][0]))       
+        if closestCenterPair is None or findPairOffset(pair) < findPairOffset(closestCenterPair):
+            closestCenterPair = pair
+            return closestCenterPair
```

Comment:
This should probably be outside of the loop, since you only want to return after checking everything.

Reply:
done; no indents